application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java

```diff
@@ -133,50 +141,60 @@ public void testTour() throws Exception
         homePage.nextStep();
         assertEquals("Title 3", homePage.getStepTitle());
         assertEquals("Step 3", homePage.getStepDescription());
-        assertFalse(homePage.hasNextStep());
+        assertTrue(homePage.hasNextStep());
         assertTrue(homePage.hasPreviousStep());
-        assertTrue(homePage.hasEndButton());
+        assertFalse(homePage.hasEndButton());
+
+        // Step 4
+        // Use a second page to test the Multipage feature
+        PageWithTour secondPage = PageWithTour.gotoPage("TourCode", "TourClass");
```

Comment:
I don't think it is the right way to test that the tour DOES redirect you.
Here you manually change the page...
Instead, I would do something like:
```
homePage.nextStep()

PageWirhTour secondPage = new PageWithTour();
assertEquals("TourClass", secondPage.getName()); // This is pseudo code ;)
```

BTW, what is the result if you visit `TourClass` first? Do you see the tour for the first time in the middle of it?